ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt

```diff
@@ -47,30 +50,30 @@ internal class VariableDataProvider(
     }
 
     fun subscriptionDuration(rcPackage: Package, locale: Locale): String? {
-        return rcPackage.product.period?.localizedPeriod(locale, formatWidth = MeasureFormat.FormatWidth.WIDE)
+        return rcPackage.product.period?.localizedPeriod(locale)
             ?: periodName(rcPackage)
     }
 
     fun introductoryOfferDuration(rcPackage: Package, locale: Locale): String? {
-        return getIntroPhaseToApply(rcPackage)?.billingPeriod?.localizedPeriod(
-            locale,
-            formatWidth = MeasureFormat.FormatWidth.WIDE,
-        )
+        return getIntroPhaseToApply(rcPackage)?.billingPeriod?.localizedPeriod(locale)
     }
 
     fun localizedPricePerPeriod(rcPackage: Package, locale: Locale): String {
         val localizedPrice = localizedPrice(rcPackage)
         return rcPackage.product.period?.let { period ->
-            var formattedPeriod = period.localizedPeriod(locale)
-            if (period.value == 1 && formattedPeriod.startsWith("1")) {
-                formattedPeriod = formattedPeriod.substring(startIndex = 1).trim()
-            }
+            val formattedPeriod = period.localizedAbbreviatedPeriod(locale)
             "$localizedPrice/$formattedPeriod"
         } ?: localizedPrice
     }
 
-    fun localizedPriceAndPerMonth(rcPackage: Package): String {
-        return "PRICE_AND_PER_MONTH"
+    fun localizedPriceAndPerMonth(rcPackage: Package, locale: Locale): String {
+        if (!rcPackage.isSubscription || rcPackage.isMonthly) {
+            return localizedPricePerPeriod(rcPackage, locale)
+        }
+        val unit = Period(1, Period.Unit.MONTH, "P1M").localizedAbbreviatedPeriod(locale)
+        val pricePerPeriod = localizedPricePerPeriod(rcPackage, locale)
+        val pricePerMonth = localizedPricePerMonth(rcPackage, locale)
+        return "$pricePerPeriod ($pricePerMonth/$unit)"
```

Comment:
not sure if we are focusing on that yet, but does this work fine for for RTL languages?

Reply:
Good question... Maybe not, but currently I just copied what we are doing in iOS. We can revisit this though

Reply:
Yeah I forgot to ask. iOS handles these strings automatically and makes them work on RTL. We'll test the Arabic paywall before launch.